staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -1446,14 +1446,29 @@ func (az *Cloud) reconcileLoadBalancer(clusterName string, service *v1.Service,
 				}
 				klog.V(10).Infof("EnsureBackendPoolDeleted(%s) for service %s: end", lbBackendPoolID, serviceName)
 
-				// Remove the LB.
-				klog.V(10).Infof("reconcileLoadBalancer: az.DeleteLB(%q): start", lbName)
-				err = az.DeleteLB(service, lbName)
+				existingLBs, err := az.ListLB(service)
```

Comment:
Instead of listing all LBs, is there no way to track the specific ID associated with the LB and calling a Get?

Reply:
I think both list and get would work well, as they all call the api once.